WeakAuras/WeakAuras.lua

```diff
@@ -4305,78 +4294,130 @@ function WeakAuras.EnsureString(input)
 end
 
 -- Handle coroutines
-local dynFrame = {};
+---@class Threads
+local threads = {
+  frame = CreateFrame("Frame"),
+  size = 0,
+  ---@type table<string, threadPriority>
+  prios = {},
+  ---@type table<string, thread>
+  instant = {},
+  ---@type table<string, thread>
+  urgent = {},
+  ---@type table<string, thread>
+  normal = {},
+  ---@type table<string, thread>
+  background = {},
+};
 do
-  -- Internal data
-  dynFrame.frame = CreateFrame("Frame");
-  dynFrame.update = {};
-  dynFrame.size = 0;
+
+  ---@alias threadPriority 'urgent' | 'normal' | 'background'
 
   -- Add an action to be resumed via OnUpdate
-  function dynFrame.AddAction(self, name, func)
-    if not name then
-      name = string.format("NIL", dynFrame.size+1);
+  ---@param name string
+  ---@param func thread
+  ---@param prio threadPriority?
+  function threads:Add(name, func, prio)
+    if not prio then
+      prio = "normal"
     end
-
-    if not dynFrame.update[name] then
-      dynFrame.update[name] = func;
-      dynFrame.size = dynFrame.size + 1
-      dynFrame.frame:Show();
+    if not self.prios[name] then
+      self.prios[name] = prio
+      self[prio][name] = func
+      self.size = self.size + 1
+      self.frame:Show()
     end
   end
 
-  -- Remove an action from OnUpdate
-  function dynFrame.RemoveAction(self, name)
-    if dynFrame.update[name] then
-      dynFrame.update[name] = nil;
-      dynFrame.size = dynFrame.size - 1
-      if dynFrame.size == 0 then
-        dynFrame.frame:Hide();
-      end
+  ---@param name string
+  ---@param prio threadPriority
+  function threads:SetPriority(name, prio)
+    if self.prios[name] then
```

Comment:
calling with an unchanged priority seems to not work, e.g. calling with "normal" while self.prios[name] is alreaedy normal will run:

self["normal"][name] = self["normal"][name]
self["normal"][name] = nil